### ISSUE TITLE
[idea] Sync opt-in status

### ISSUE THREAD

**Issue by rcknr** [mailchimp/Firebase#14]
A boolean field to sync user opt-in status (subscribed/unsubscribed).

**Comment by barticus** [comment]
Closing, lets use the PR as the discussion point (as I've commented there!)

**Comment by rcknr** [comment]
Hi! I have submitted this while working on a project that is now concluded. However, I could modify the PR to accommodate your suggestions. I'm not sure how to proceed since you have closed the request. Please advice.

**Comment by barticus** [comment]
Hi @rcknr , sorry for the confusion, I wanted to make sure we have a single thread for the discussion, and would rather that be on your PR https://github.com/mailchimp/Firebase/pull/15, if you're happy with that? Alternatively, can convert this issue to a feature one, and link the PR.
If you'd like to do the changes, that would be great, thanks!

**Comment by barticus** [comment]
Similar to as proposed in #43 , I think we can have a object based syntax to allow syncing of contact status in the merge fields handler, as that handler talks to the same API as where status updates need to be sent.

Merge field config would then be in this shape:
```json
{
      "mergeFields": {
        "firstName": "FNAME",
        "lastName": "LNAME",
        "phoneNumber": "PHONE"
      },
      "status": {
        "mailchimpFieldName": "subscriberStatus", <- Field that will indicate the users Status
        "format": "boolean|string" <- Determines whether the Status field will be interpreted as a simple "subscribed/unsubscribed" or whether they want to provide the full status
      }
      "subscriberEmail": "emailAddress"
    } 
```
With the status configuration being optional.

**Comment by barticus** [comment]
This has been implemented in #57
